### PR TITLE
[Feature] Restrict the format of variable code in the variable management module #1851

### LINKED ISSUE #1851: [Feature] Restrict the format of variable code in the variable management module
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

- variable code length should be no less than 3 and no more than 50 characters
- variable code can only contain letters, numbers, middle bars, bottom bars and dots, and the beginning can only be letters, For example, kafka_cluster.brokers-520

### Usage Scenario

_No response_

### Related issues

- [Feature] Add basic functionality of variable modules such as create, modify, delete, query and search #1779 
- [Proposal][Feature] Add variable management module #1477 

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by muchunjin**
## What changes were proposed in this pull request

Issue Number: close #1851  

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

File: streampark-console/streampark-console-webapp/src/views/system/variable/Add.vue [hunk 1]
```diff
@@ -127,23 +127,21 @@ export default {
...
-            }
-          })
-        }
+        this.validateStatus = 'validating'
+        checkVariableCode({
+          variableCode: variableCode
+        }).then((resp) => {
+          if (resp.status !== 'success') {
+            this.validateStatus = 'error'
+            this.help = resp.message
```
Comment (1996fanrui) [inline streampark-console/streampark-console-webapp/src/views/system/variable/Add.vue]: The check can be added in front-end. You can refer how to limit the User name length.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/VariableController.java [hunk 2]
```diff
@@ -87,6 +92,11 @@ public RestResponse deleteVariables(@Valid Variable variable) throws Exception {
 
     @PostMapping("check/code")
     public RestResponse checkVariableCode(@RequestParam Long teamId, @NotBlank(message = "{required}") String variableCode) {
+        try {
+            this.checkVariableCodeFormat(variableCode);
+        } catch (ApiAlertException e) {
+            return RestResponse.fail(e.getMessage(), ResponseCode.CODE_FAIL_ALERT);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/VariableController.java:98]: we don't need try exception there, just `throw exception`

**Comment by muchunjin** [comment]
@1996fanrui @wolfboys 
I modified three places
- Added the inspection of variable code format in the front end.
- Moved the variable management module to core and adjusted the order of the menus.
- Fixed menu not sorting when admin is logged in.

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**